File: src/main/java/org/apache/fineract/client/services/ReportsApi.java
/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements. See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership. The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
package org.apache.fineract.client.services;

import org.apache.fineract.client.CollectionFormats.*;

import rx.Observable;
import retrofit2.http.*;

import okhttp3.RequestBody;
import okhttp3.ResponseBody;
import okhttp3.MultipartBody;

import org.apache.fineract.client.models.DeleteReportsResponse;
import org.apache.fineract.client.models.GetReportsResponse;
import org.apache.fineract.client.models.GetReportsTemplateResponse;
import org.apache.fineract.client.models.PostReportsResponse;
import org.apache.fineract.client.models.PostRepostRequest;
import org.apache.fineract.client.models.PutReportRequest;
import org.apache.fineract.client.models.PutReportResponse;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public interface ReportsApi {
  /**
   * Create a Report
   * 
   * @param postRepostRequest  (required)
   * @return Observable&lt;PostReportsResponse&gt;
   */
  @Headers({
    "Content-Type:application/json"
  })
  @POST("reports")
  Observable<PostReportsResponse> createReport(
    @retrofit2.http.Body PostRepostRequest postRepostRequest
  );

  /**
   * Delete a Report
   * Only non-core reports can be deleted.
   * @param id id (required)
   * @return Observable&lt;DeleteReportsResponse&gt;
   */
  @DELETE("reports/{id}")
  Observable<DeleteReportsResponse> deleteReport(
    @retrofit2.http.Path("id") Long id
  );

  /**
   * Retrieve Report Template
   * This is a convenience resource. It can be useful when building maintenance user interface screens for client applications. The template data returned consists of any or all of:  Field Defaults Allowed description Lists  Example Request :   reports/template
   * @return Observable&lt;GetReportsTemplateResponse&gt;
   */
  @GET("reports/template")
  Observable<GetReportsTemplateResponse> retrieveOfficeTemplate();
    

  /**
   * Retrieve a Report 
   * Example Requests:  reports/1   reports/1?template&#x3D;true
   * @param id id (required)
   * @return Observable&lt;GetReportsResponse&gt;
   */
  @GET("reports/{id}")
  Observable<GetReportsResponse> retrieveReport(
    @retrofit2.http.Path("id") Long id
  );

  /**
   * List Reports
   * Lists all reports and their parameters.  Example Request:  reports
   * @return Observable&lt;List&lt;GetReportsResponse&gt;&gt;
   */
  @GET("reports")
  Observable<List<GetReportsResponse>> retrieveReportList();
    

  /**
   * Update a Report
   * Only the useReport description can be updated for core reports.
   * @param id id (required)
   * @param putReportRequest  (required)
   * @return Observable&lt;PutReportResponse&gt;
   */
  @Headers({
    "Content-Type:application/json"
  })
  @PUT("reports/{id}")
  Observable<PutReportResponse> updateReport(
    @retrofit2.http.Path("id") Long id, @retrofit2.http.Body PutReportRequest putReportRequest
  );

}

File: src/main/kotlin/org/openapitools/client/apis/ReportsApi.kt
package org.openapitools.client.apis

import org.openapitools.client.models.DeleteReportsResponse
import org.openapitools.client.models.GetReportsResponse
import org.openapitools.client.models.GetReportsTemplateResponse
import org.openapitools.client.models.PostReportsResponse
import org.openapitools.client.models.PostRepostRequest
import org.openapitools.client.models.PutReportRequest
import org.openapitools.client.models.PutReportResponse
import retrofit2.http.Body
import retrofit2.http.DELETE
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.PUT
import retrofit2.http.Path

interface ReportsApi {
    /**
     * Create a Report
     *
     * Responses:
     *  - 200: OK
     *
     * @param postRepostRequest
     * @return [PostReportsResponse]
     */
    @POST("v1/reports")
    suspend fun createReport(@Body postRepostRequest: PostRepostRequest): PostReportsResponse

    /**
     * Delete a Report
     * Only non-core reports can be deleted.
     * Responses:
     *  - 200: OK
     *
     * @param id id
     * @return [DeleteReportsResponse]
     */
    @DELETE("v1/reports/{id}")
    suspend fun deleteReport(@Path("id") id: Long): DeleteReportsResponse

    /**
     * Retrieve Report Template
     * This is a convenience resource. It can be useful when building maintenance user interface screens for client applications. The template data returned consists of any or all of:  Field Defaults Allowed description Lists  Example Request :   reports/template
     * Responses:
     *  - 200: OK
     *
     * @return [GetReportsTemplateResponse]
     */
    @GET("v1/reports/template")
    suspend fun retrieveOfficeTemplate(): GetReportsTemplateResponse

    /**
     * Retrieve a Report
     * Example Requests:  reports/1   reports/1?template&#x3D;true
     * Responses:
     *  - 200: OK
     *
     * @param id id
     * @return [GetReportsResponse]
     */
    @GET("v1/reports/{id}")
    suspend fun retrieveReport(@Path("id") id: Long): GetReportsResponse

    /**
     * List Reports
     * Lists all reports and their parameters.  Example Request:  reports
     * Responses:
     *  - 200: OK
     *
     * @return [kotlin.collections.List<GetReportsResponse]
     */
    @GET("v1/reports")
    suspend fun retrieveReportList(): List<GetReportsResponse>

    /**
     * Update a Report
     * Only the useReport description can be updated for core reports.
     * Responses:
     *  - 200: OK
     *
     * @param id id
     * @param putReportRequest
     * @return [PutReportResponse]
     */
    @PUT("v1/reports/{id}")
    suspend fun updateReport(
        @Path("id") id: Long,
        @Body putReportRequest: PutReportRequest
    ): PutReportResponse

}
